### DISCUSSION TITLE
lazy/code-split implementation

### DISCUSSION THREAD

**Discussion by akhilshastri** [pmndrs/valtio#772] [Q&A]
is there a way to have lazy/code-split implementation like - redux has injectReducer

**Comment by dai-shi** [comment]
Valtio (and Jotai) is good at code splitting from the start.
You can just define multiple state separately and if you want to combine, you can do that later.
(TS types are a little bit tricky, but should be doable.)

https://github.com/pmndrs/valtio/blob/734c24db74fbc6cf35d63b939f67b17602dbfe1e/docs/how-tos/how-to-split-and-compose-states.mdx#L25
https://github.com/pmndrs/valtio/blob/734c24db74fbc6cf35d63b939f67b17602dbfe1e/docs/how-tos/how-to-organize-actions.mdx#L12
